eiger_io/fs_handler.py

```diff
@@ -20,3 +21,53 @@ def __call__(self, seq_id):
         master_path = '{}_{}_master.h5'.format(self._base_path, seq_id)
         # TODO Return a multi-dimensional PIMS seq.
         return np.array(EigerImages(master_path))
+
+
+class FixedEigerImages(EigerImages):
+    def __init__(self, path, metadata):
+        super().__init__(path)
+        self._metadata = metadata
+
+    @property
+    def md(self):
+        return self._metadata
+
+    @property
+    def dtype(self):
+        return self.pixel_type
+
+    @property
+    def shape(self):
+        return self.frame_shape
+
+
+class LazyEigerHandler(HandlerBase):
+    specs = {'AD_EIGER'} | HandlerBase.specs
+    def __init__(self, fpath, frame_per_point, mapping=None):
+        # create pims handler
+        self.vals_dict = EIGER_MD_DICT.copy()
+        if mapping is not None:
+            self.vals_dict.update(mapping)
+        self._base_path = fpath
+        self.fpp = frame_per_point
+
+    def __call__(self, seq_id):
+        master_path = '{}_{}_master.h5'.format(self._base_path, seq_id)
+        md = {}
+        print('hdf5 path = %s' % master_path)
+        with h5py.File(master_path, 'r') as f:
+            md = {k: f[v].value for k, v in self.vals_dict.items()}
+        # the pixel mask from the eiger contains:
+        # 1  -- gap
+        # 2  -- dead
+        # 4  -- under-responsive
+        # 8  -- over-responsive
+        # 16 -- noisy
+        pixel_mask = md['pixel_mask']
```

Comment:
@sameera2004 What you just described is not the point of this code.  I was not around for any discussion of data type problems.  

This is the problem, as I understand it:

The pixel mask that comes from the detector has a number of nonzero values.  The meaning of those values is enumerated in the comments section above.  What CHX wanted was to convert this multi-valued pixel mask into a mask where `1` means that the pixel is good and `0` means that the pixel is bad.  CHX wanted this mask to be 0 or 1 because they were simply multiplying masks to combine masks.  I wrote this on the fly with yugang and andrei sitting over my shoulder.  This should probably just be rewritten to be this:

```python
pixel_mask = ~(md['pixel_mask'] == 0)
md['pixel_mask'] = pixel_mask.astype(int)
```